practice.rb

```diff
@@ -0,0 +1,50 @@
+## Problem #1
+def print_array(array_of_integers)
+  print array_of_integers
+end
+
+## Problem #2
+def stretch(array)
+  stretch = []
+  array.each do |i|
+    if i % 2 == 0
+      2.times do
+        stretch << i / 2
+      end
+    else
+      stretch << (i / 2) + 1
+      stretch << (i / 2)
+    end
+  end
+  return stretch
+end
+
+## Problem #3
+def numUnique(integer_array)
+  numbers = {}
+  integer_array.each do |input|
+    if !numbers.has_key?(input)
+      numbers[input] = 1
+    end
+  end
+  return numbers.length
+end
```

Comment:
This works, but it basically just utilizes the hash as a counter, doesn't make use of the properties of a hash. 

What I was thinking was something more where if you had:
[1,1,1,4,5,5]

you'd create a hash
{ 1 => 3, 4 => 1, 5 => 2}

And then you could just get the size of the hash for the unique elements, but you also have created for yourself information if we were to ask a different, but related question, like how many values only appear once in the array, etc